### PR TITLE
Added GPS to ZumoComSystem

### PR DESCRIPTION
**Pull request by gabryelreyes**

### CODE REVIEW

File: doc/architecture/uml/LogicalView/HALSim.plantuml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    class Board << namespace >> {
+        + getButton() : IButton&
+        + getBattery() : IBattery&
+        + getRobot() : IRobot&
+        + getNetwork() : INetwork&
+        + getLedRed() : ILed&
+        + getLedGreen() : iLed&
+        + getLedBlue() : ILed&
+        + getGps() : IGps&
```
Comment (BlueAndi) [inline doc/architecture/uml/LogicalView/HALSim.plantuml:15]: Isn't it a pointer?

File: doc/architecture/uml/LogicalView/HALSim.plantuml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    }
+}
+
+Robot *--> WebotsSerialDrv
+Network *--> NetworkLib
+
+WebotsSerialDrv *--> Stream
+WebotsSerialDrv *--> WebotsEmitter
+WebotsSerialDrv *--> WebotsReceiver
+Gps *--> WebotsGps
```
Comment (BlueAndi) [inline doc/architecture/uml/LogicalView/HALSim.plantuml:96]: GPS is optional

File: lib/HALInterfaces/src/IGps.h [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    /**
+     * Get the current position coordinates in millimeters.
+     *
+     * @param[out] xPos  The X position in mm.
+     * @param[out] yPos  The Y position in mm.
+     *
+     * @return If the position was successfully retrieved.
+     */
+    virtual bool getPosition(int32_t& xPos, int32_t& yPos) = 0;
```
Comment (BlueAndi) [inline lib/HALInterfaces/src/IGps.h:75]: Heading is not part?

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: lib/HALInterfaces/src/IGps.h (hunk 3 above)
Comment (gabryelreyes) [reply lib/HALInterfaces/src/IGps.h:75]: No, heading would be done with a compass, not a GPS

Comment (BlueAndi) [reply lib/HALInterfaces/src/IGps.h:75]: I am asking because the odo interface contains always both, position and heading.

Comment (gabryelreyes) [reply lib/HALInterfaces/src/IGps.h:75]: Yes definitely, but I would implement everything component-wise. First GPS, then Compass, and then the switch between both "inputs" for the calculation of the position

File: doc/architecture/uml/LogicalView/HALSim.plantuml (hunk 2 above)
Comment (gabryelreyes) [reply doc/architecture/uml/LogicalView/HALSim.plantuml:96]: The use of the GPS is optional at the application level but the GPS Class in the HAL requires the Webots GPS class.

File: lib/ConvoyFollower/src/App.cpp [hunk 4]
```diff
@@ -635,6 +637,18 @@ void App_currentVehicleChannelCallback(const uint8_t* payload, const uint8_t pay
         Telemetry data(currentVehicleData->xPos, currentVehicleData->yPos, currentVehicleData->orientation,
                        currentVehicleData->left, currentVehicleData->right, currentVehicleData->center, proximity);
 
+        if (gps != nullptr)
```
Comment (BlueAndi) [inline lib/ConvoyFollower/src/App.cpp:640]: Please comment here, that overwritting is intended and why.

File: lib/ConvoyLeader/src/App.cpp [hunk 5]
```diff
@@ -615,6 +617,18 @@ void App_currentVehicleChannelCallback(const uint8_t* payload, const uint8_t pay
         Telemetry data(currentVehicleData->xPos, currentVehicleData->yPos, currentVehicleData->orientation,
                        currentVehicleData->left, currentVehicleData->right, currentVehicleData->center, proximity);
 
+        if (gps != nullptr)
+        {
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/App.cpp:621]: Please comment here, that overwritting is intended and why.

**Review by BlueAndi — CHANGES_REQUESTED** [review]

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**